### PR TITLE
Send error event when service is down

### PR DESCRIPTION
**Pull request by ruflin**
If a service does not respond with stats requested by a metricset, an error is sent to elasticsearch instead. This allows to also track and monitor errors of systems.

### CODE REVIEW

File: metricbeat/helper/module.go [hunk 1]
```diff
@@ -162,13 +162,18 @@ func (m *Module) FetchMetricSets(b *beat.Beat, metricSet *MetricSet) {
...
-		// TODO: Still publish event with error
-		return
-	}
+		errorMessage := fmt.Sprintf("Fetching events for MetricSet %s in Module %s returned error: %s", metricSet.Name, m.name, err)
+		logp.Err(errorMessage)
 
-	events, err = m.processEvents(events, metricSet)
+		// Publish event with error
+		event := common.MapStr{
+			"error": fmt.Sprintf("%s", err),
```
Comment (andrewkroh) [inline metricbeat/helper/module.go]: No need to `Sprintf`, just call [err.Error()](https://golang.org/pkg/builtin/#error).

File: metricbeat/helper/module.go [hunk 2]
```diff
@@ -162,13 +162,18 @@ func (m *Module) FetchMetricSets(b *beat.Beat, metricSet *MetricSet) {
 	events, err := metricSet.Fetch()
 
 	if err != nil {
-		// TODO: Also list module?
-		logp.Err("Fetching events in MetricSet %s returned error: %s", metricSet.Name, err)
-		// TODO: Still publish event with error
-		return
-	}
+		errorMessage := fmt.Sprintf("Fetching events for MetricSet %s in Module %s returned error: %s", metricSet.Name, m.name, err)
```
Comment (andrewkroh) [inline metricbeat/helper/module.go]: Why not remove the `Sprintf` and just call `logp.Err()` directly with the same arguments?

Comment (ruflin) [inline metricbeat/helper/module.go]: That is a left over from my first implementation, where I used to variable also for the error message below. But the metricset and module are not needed there, as it is already clear from the event. Changed.

File: metricbeat/helper/module.go (hunk 1 above)
Comment (ruflin) [inline metricbeat/helper/module.go]: Fixed

**Pull request merged**